DevelopGitHub/GitHubExtension/GitHubExtension.Notes.WebApi/Services/NoteService.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System.Data.Entity;
+using System.Linq;
+using System.Threading.Tasks;
+using GitHubExtension.Notes.DAL.Model;
+using GitHubExtension.Notes.WebApi.ViewModels;
+using AutoMapper.QueryableExtensions;
+
+namespace GitHubExtension.Notes.WebApi.Services
+{
+    public class NoteService : INotesService
+    {
+        private readonly NoteContext notesContext;
+
+        public NoteService(NoteContext notesContext)
+        {
+            this.notesContext = notesContext;
+        }
+
+        public async Task<AddNoteModel> GetNote(int noteId)
+        {
+            var note = await notesContext.Notes
+                .Where(x => x.Id == noteId)
+                .ProjectTo<AddNoteModel>()
+                .FirstOrDefaultAsync();
+            return note;
+        }
+
+        public async Task<Note> AddNote(AddNoteModel addNote)
+        {
+            var note = new Note()
```

Comment:
Thought it is better to move that process from Controller to Service. Am I wrong and it is a better practice to pass an entity to service?